XIPs/xip-reaction-content-type.md

```diff
@@ -0,0 +1,60 @@
+---
+xip: 
+title: Reaction Content Type
+description: Content type for reactions
+author: Kunal Mondal
+status: Draft 
+type: Standards Track
+category: XRC
+created: 2023-03-24
+---
+
+## Abstract
+
+This XRC proposes a new content type for messages that enriches the messaging experience and let's users express themselves using emojis. It attempts to include only the bare minimum for clients to determine how to display such messages in order to provide flexibility for more rich types in the future. 
+
+## Motivation
+
+When it comes to having a comfortable conversation, reacting to messages with emojis is necessary as it gives users the freedom to express their messages in form of emoji. In order to provide users a real messaging platform, XMTP needs to provide this capability.
+
+## Specification
+
+Proposed content type:
+
+```js
+{
+    authorityId: "xmtp.org",
+    typeId: "reaction",
+    versionMajor: 1,
+    versionMinor: 0,
+}
+```
+
+The reaction MUST include the following parameters:
+
+Adding info about the message we’re reacting to (it’s message id, sender address, message content, content type id and the reaction). 
+
+```ts
+{
+    messageId: string
+    senderAddress: string
+    messageContent: string
```

Comment:
Would this be the `message.content` from the message you are reacting to? Why do we need to duplicate that in the Reaction? Couldn't clients just read the original message?

Reply:
yes `messageContent` is the `message.content` from the message we're replying to. 

It's necessary to duplicate it for the following use cases : 

1. We're loading the last messages of all the conversations of a user to show them in the chat previews. Whenever the last message is a Reaction, we need the `messageContent` and `contentTypeId` to display which message it has been reacted to. ( See the image)

![image](https://user-images.githubusercontent.com/89394181/227577384-9451720b-c757-4ca0-9508-e091d35b4e27.png)

2. There are cases where the message we reacted to is not loaded yet (due to the pagination) and so we won't have `messageContent` and `contentTypeId` info if we try to filter and find it from the message's state in the application through the Reaction's messageId.

Reply:
I think we should just pass IDs here, since duplication could lead to different content being passed in different places which would lead to inconsistency. It also doesn't account for non-text messages.

Reply:
Yes we can remove the `messageContent `and `contentTypeId` and only keep the `messageId` and `reaction` for this to work.

Although as we won't have the the content of the message we reacted to, we can't show them in the preview. It will be just the reaction.

![image](https://user-images.githubusercontent.com/89394181/228048516-50f69ec8-c3cb-4ab4-b833-cd431714c6cf.png)